galasa-inttests-parent/dev.galasa.inttests/src/main/java/dev/galasa/inttests/docker/AbstractDockerUbuntuLocal.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright contributors to the Galasa project
+ */
+package dev.galasa.inttests.docker;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+
+import dev.galasa.BeforeClass;
+import dev.galasa.ResultArchiveStoreContentType;
+import dev.galasa.SetContentType;
+import dev.galasa.core.manager.StoredArtifactRoot;
+
+public abstract class AbstractDockerUbuntuLocal extends AbstractDocker {
+	
+	@StoredArtifactRoot
+    public Path storedArtifactRoot;
+	
+	@BeforeClass
+	public void setupEnvironment() throws Exception {
+		shell = getDockerLinuxImage().getCommandShell();
+		
+		logger.info("Checking if docker already exists");
+		String res = shell.issueCommand("sudo docker ps");
+		if(res.contains("CONTAINER ID")) {
+			logger.info("Docker is already installed. Skipping install step.");
+		} else {
+			installDocker();
+		}
+			
+		logger.info("Exposing Docker engine");
+		Path servicePath = getDockerLinuxImage().getRoot();
+		servicePath = servicePath.resolve("lib/systemd/system/docker.service");
+		Path homePath = getDockerLinuxImage().getHome();
+		homePath = homePath.resolve("docker.service");
+
+		List<String> newLines = new ArrayList<>();
+		for (String line : Files.readAllLines(servicePath, StandardCharsets.UTF_8)) {
```

Comment:
Too complicated.   Use IOUtils to read the while file into a single String.
You are not checking to see if the line has already been changed.
simply check the string contains tcp://0.0.0.0:
if not replace the string -H fd:// with -H fd:// -H://0.0.0.0:2376